electron/launcher/main-process/helpers/login.js

```diff
@@ -0,0 +1,203 @@
+    .then(cookies => {
+      let promises = [];
+      let expiry = 0;
+
+      promises.push(keytar.setPassword('Dietstory', body.name, encryptedPassword));
+      for(cookieObject of cookies){
+        if(cookieObject.hasOwnProperty('csrftoken')){
+          csrfToken = cookieObject.csrftoken;   // GLOBAL VARIABLE
+          promises.push(keytar.setPassword('Dietstory_CSRF', body.name, cookieObject.csrftoken));
+        }
+        else if(cookieObject.hasOwnProperty('sessionid')){
+          sessionKey = cookieObject.sessionid;  // GLOBAL VARIABLE
+          promises.push(keytar.setPassword('Dietstory_Session', body.name, cookieObject.sessionid));
+          expiry = Date.parse(cookieObject.expires);
+        }
+      }
+      
+      return Promise.all(promises).then(() => {
+        return expiry;
+      })
+      .catch(error => {
+        throw error;
+      });
+    })
+    // SAVE USER INFO FOR FUTURE AUTO-LOGIN TO USE (ENCRYPTED)
+    .then(expiry => {
+      let encrypt = string => {
+        return new Promise((resolve, reject) => {
+          let cipher = crypto.createCipheriv(secret.algorithm, Buffer.from(secret.key, 'hex'), Buffer.from(secret.iv, 'hex'));
+          let encrypted = cipher.update(string, 'utf8', 'hex');
+          encrypted += cipher.final('hex');
+          resolve(encrypted);
+        });
+      }
+
+      sessionExpiry = expiry;   // GLOBAL VARIABLE
+      let userInfo = {...body, session_expiry: expiry};
+
+      return encrypt(JSON.stringify(userInfo)).then(encyrptedData => {
+        fs.writeFileSync(path.join(app.getPath('userData'), 'user_info'), encyrptedData, 'utf-8');
+        return userInfo;
+      })
+      .catch(error => {
+        throw error;
+      });
+    })
+    // RETURNS DJANGO RESPONSE BODY
+    .then(userInfo => {
+      resolve({body: body, userInfo: userInfo});
+    })
+    .catch(error => {
+      reject(error);
+    });
+  });
+}
+
+
+
+
+
+
+// AUTO LOGIN
+const autoLogin = () => {
+  return new Promise((resolve, reject) => {
+    // READ AND DECRYPT user_info FILE
+    fs.readFile(path.join(app.getPath('userData'), 'user_info'), 'utf-8').then(encryptedData => {
+      let decrypt = data => {
+        return new Promise((resolve, reject) => {
+          let encrypted = Buffer.from(data, 'hex');
+          let decipher = crypto.createDecipheriv(secret.algorithm, Buffer.from(secret.key, 'hex'), Buffer.from(secret.iv, 'hex'));
+          let decrypted = decipher.update(encrypted);
+          decrypted = Buffer.concat([decrypted, decipher.final()]);
+          resolve(decrypted.toString());
+        })
+      }
+
+      return decrypt(encryptedData).then(userInfo => {
+        return JSON.parse(userInfo);
+      })
+    })
+    // GET USER'S PASSWORD
+    .then(userInfo => {
+      return keytar.getPassword('Dietstory', userInfo.name).then(password => {
+        return aes256.decrypt(password).then(decryptedPassword => {
```

Comment:
What happens when `autoLogin` is called when this key pair doesn't exist?

Reply:
Does it error out or return a `null`

Reply:
It throws an error which is ignored. To the user, they will simply remain on the login screen and must re-authenticate manually.